pkg/reconciler/v1alpha1/revision/reconcile_resources.go

```diff
@@ -188,13 +188,12 @@ func (c *Reconciler) reconcileService(ctx context.Context, rev *v1alpha1.Revisio
 		// It may change if a user edits things around our controller, which we
 		// should not allow, or if our expectations of how the service should look
 		// changes (e.g. we update our controller with new sidecars).
-		var changed Changed
-		_, changed, err = c.checkAndUpdateService(ctx, rev, resources.MakeK8sService, service)
+		_, changed, err := c.checkAndUpdateService(ctx, rev, resources.MakeK8sService, service)
 		if err != nil {
 			logger.Errorf("Error updating Service %q: %v", serviceName, err)
 			return err
 		}
-		if changed == WasChanged {
+		if changed == wasChanged {
 			logger.Infof("Updated Service %q", serviceName)
 			rev.Status.MarkDeploying("Updating")
```

Comment:
Remind me again why we want to update the revision status when UpdateService happens but not UpdateDeployment?

They both have the desired behavior of:
1.) Fix accidental user modification
2.) Update sidecars when Knative is updated.

Reply:
Per Matt: we only care if endpoints changed, which will be seen when we reconcile the service (i.e. if deployment changes were such that endpoints didn't change, we don't care to update the status)